expr.go

```diff
@@ -334,23 +377,40 @@ func getFloatArg(e *expr, n int) (float64, error) {
 		return 0, ErrMissingArgument
 	}
 
-	if e.args[n].etype != etConst {
-		return 0, ErrBadType
+	return doGetFloatArgDefault(e.args[n], 0)
+}
+
+func getFloatArgDefault(e *expr, n int, v float64) (float64, error) {
+	if len(e.args) <= n {
+		return v, nil
 	}
 
-	return e.args[n].val, nil
+	return doGetFloatArgDefault(e.args[n], v)
 }
 
-func getFloatArgDefault(e *expr, n int, v float64) (float64, error) {
+func getFloatNamedOrPosArgDefault(e *expr, k string, n int, v float64) (float64, error) {
 	if len(e.args) <= n {
 		return v, nil
 	}
 
-	if e.args[n].etype != etConst {
+	if a := getNamedArg(e.args, k); a != nil {
+		return doGetFloatArgDefault(a, v)
+	}
+
+	return getFloatArgDefault(e, n, v)
+}
+
+func doGetFloatArgDefault(e *expr, v float64) (float64, error) {
+	if e.etype != etConst && e.etype != etKV {
 		return 0, ErrBadType
 	}
 
-	return e.args[n].val, nil
+	f := e.val
+	if f == 0 {
```

Comment:
This would make `floatArg=0`, when also not the default such as in `transfromNull()`, impossible to pass. I'm not sure that's a problem or not - but it is new.